packages/backend/src/workers/provider/LlamaCppPython.ts

```diff
@@ -254,8 +275,29 @@ export class LlamaCppPython extends InferenceProvider {
       case VMType.LIBKRUN_LABEL:
         return gpu ? llamacpp.vulkan : llamacpp.default;
       // no GPU support
+      case VMType.UNKNOWN:
+        return this.isNvidiaCDIConfigured(gpu) ? llamacpp.cuda : llamacpp.default;
       default:
         return llamacpp.default;
     }
   }
+
+  protected isNvidiaCDIConfigured(gpu?: IGPUInfo): boolean {
+    // NVIDIA cdi must be set up to use GPU acceleration on Linux.
+    // Check the known locations for the configuration file
+    const knownLocations = [
+      '/etc/cdi/nvidia.yaml', // Fedora
+    ];
+
+    if (gpu?.vendor !== GPUVendor.NVIDIA) return false;
+
+    let cdiSetup = false;
+    for (const location of knownLocations) {
+      if (fs.existsSync(location)) {
```

Comment:
This would only work on Linux, which may not be directly an issue, but I would like to have a safeguard to this function.
 
```
If(!env.isLinux) throw new Error('cannot check nvidia CDI on non-Linux platform')
```